cdap-api/src/main/java/co/cask/cdap/api/dataset/DatasetDefinition.java

```diff
@@ -71,24 +74,28 @@
    * Provides dataset admin to be used to perform administrative operations on the dataset instance defined by passed
    * {@link DatasetSpecification}.
    *
-   * @param spec specification of the dataset instance.
+   *
+   * @param datasetContext context for the dataset
    * @param classLoader classloader to use when executing admin operations
+   * @param spec specification of the dataset instance.
    * @return dataset admin to perform administrative operations
    * @throws IOException
    */
-  A getAdmin(DatasetSpecification spec, ClassLoader classLoader) throws IOException;
+  A getAdmin(DatasetContext datasetContext, ClassLoader classLoader, DatasetSpecification spec) throws IOException;
```

Comment:
Must have happened during refactoring in the IDE. I'll move it to be the second parameter.